vernac/record.ml

```diff
@@ -582,6 +583,17 @@ let extract_record_data records =
   in
   ps, data
 
+(* Returns [None] for regular records, [Some false] for regular typeclasses
+   (records) and [Some true] for definitional typeclasses. *)
+let kind_class =
+  let open Vernacexpr in
+  function Class def -> Some def
+  | Inductive_kw | CoInductive | Variant | Record | Structure -> None
+
+let implicits_of_context ctx =
+  List.map (fun name -> CAst.make (Some (name,true)))
+    (List.rev (Anonymous :: (List.map RelDecl.get_name ctx)))
+
 let pre_process_structure udecl kind ~template ~cumulative ~poly ~primitive_proj finite (records : Ast.t list) =
```

Comment:
Let's take this occassion to document this function.

Reply:
Unfortunately, I don't master the details of this function (only moved it around). I'd rather not write anything rather than a misleading comment.